### DISCUSSION TITLE
SecOnion 2.3.2 - Kibana Error:404 - Elastic seems to have stopped

### DISCUSSION THREAD

**Discussion by Git-Me-Some-Hub** [Security-Onion-Solutions/securityonion#3634] [Unsupported Versions] [answered]
So I have been battling this for a bit now, and after a few times of blowing away the system and starting a new I cannot get past the problem of elastic dying after about 1-2 weeks of SPAN/Beat data coming in. I will attach some logs that are just repeating, but don't mean a ton to me. 

I was seeing the Kibana web server say it was still waiting to start up, tried a workaround listed [here](https://github.com/Security-Onion-Solutions/securityonion/discussions/2619) -- [here ](https://docs.securityonion.net/en/2.3/kibana.html#diagnostic-logging ) but it did not work. Now I am just seeing a 404 error, and looking at the Kibana logs it says it cannot connect to Elastic. The service show online, though. Other times it does show an Error State. I have tried to restart Elasticsearch a couple times, but does not seem to continue logging. It has been over two days since any logs were created. 

It is a VMware Esxi machine with 4 cores, and 16 GB of RAM. I have had it on a WD Red 5400rpm drive for the test machine. All had been working on previous version, I cannot remember if it was maybe around version  2.0? Anyway, reading the documentation about it being in RAID10, I bumped it up to a 7200rpm Seagate drive. 

Definitely at a loss. Any help is appreciated. 

[Elastic_Error00.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6209522/Elastic_Error00.txt)
[nl-seconion6t01.log](https://github.com/Security-Onion-Solutions/securityonion/files/6209528/nl-seconion6t01.log)

A few snip its: 
`[WARN ][rest.suppressed          ] path: /elastalert_status/_search, params: {size=1000, index=elastalert_status}
org.elasticsearch.action.search.SearchPhaseExecutionException: all shards failed`

`{"type":"log","@timestamp":"2021-03-18T03:00:49Z","tags":["warning","savedobjects-service"],"pid":292,"message":"Another Kibana instance appears to be migrating the index. Waiting for that migration to complete. If no other Kibana instance is attempting migrations, you can get past this message by deleting index .kibana_2 and restarting Kibana."}
`

**Comment by TOoSmOotH** [comment, marked as answer]
https://github.com/Security-Onion-Solutions/securityonion/discussions/3639

**Reply by Git-Me-Some-Hub** [reply]
Thank for the reply but this is the exact same "fix" I posted in the links above. I tried this in the past and it did not fix anything. I am not seeing ElasticSearch showing an Error and Kibana has nothing in the log files. 
This was that link post: https://github.com/Security-Onion-Solutions/securityonion/discussions/2619

**Reply by Git-Me-Some-Hub** [reply]
Since Elastic was also not logging any more, I rebooted the server and now I do get that Kibana error log again, but when I try the curl command I get an 
ssl error: `SSL received a record that exceeded the maximum permissible length.`
Command: `curl -k -XDELETE https://localhost:9200/.kibana_2`

Logs: `{"type":"log","@timestamp":"2021-04-01T03:34:15Z","tags":["info","savedobjects-service"],"pid":292,"message":"Waiting until all Elasticsearch nodes are compatible with Kibana before starting saved objects migrations..."}
{"type":"log","@timestamp":"2021-04-01T03:34:15Z","tags":["info","savedobjects-service"],"pid":292,"message":"Starting saved objects migrations"}
{"type":"log","@timestamp":"2021-04-01T03:34:15Z","tags":["info","savedobjects-service"],"pid":292,"message":"Creating index .kibana_2."}
{"type":"log","@timestamp":"2021-04-01T03:34:16Z","tags":["error","elasticsearch","data"],"pid":292,"message":"[resource_already_exists_exception]: index [.kibana_2/sZHlztyOQRi6GKGvPBsl5w] already exists"}
{"type":"log","@timestamp":"2021-04-01T03:34:16Z","tags":["warning","savedobjects-service"],"pid":292,"message":"Unable to connect to Elasticsearch. Error: resource_already_exists_exception"}
{"type":"log","@timestamp":"2021-04-01T03:34:16Z","tags":["warning","savedobjects-service"],"pid":292,"message":"Another Kibana instance appears to be migrating the index. Waiting for that migration to complete. If no other Kibana instance is attempting migrations, you can get past this message by deleting index .kibana_2 and restarting Kibana."}
`

**Reply by Git-Me-Some-Hub** [reply]
So I did a bit more looking around and did not see anything, so I decided to try the curl command without the S in HTTPs. When I performed it with just HTTP it seems to have worked. Small thing I suppose I should have tried, but is the documentation suppose to be HTTP, or should it work with HTTPS as well?
